.build/run-ci.sh

```diff
@@ -0,0 +1,157 @@
+            ;;
+        -ctx|--kubecontext)
+            unset KUBECONFIG
+            KUBECONTEXT="$2" # This sets the KUBECONTEXT variable to the next argument.
+            shift            # This shifts the arguments to the left, discarding the current argument and moving to the next one.
+            shift            # This is an additional shift to move to the argument after the option value.
+            ;;
+        --include-test-stage)
+            INCLUDE_TEST_STAGE="$2"
+            shift
+            shift
+            ;;
+        *)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$KUBECONFIG" ] && [ -z "$KUBECONTEXT" ]; then
+    echo "Please provide either the path to the kubeconfig using -c|--kubeconfig option or the kubecontext using -ctx|--kubecontext option."
+    exit 1
+fi
+
+# This sets the kubeconfig and kubecontext if provided
+if [ -n "$KUBECONFIG" ]; then
+    export KUBECONFIG="$KUBECONFIG"
+fi
+
+if [ -n "$KUBECONTEXT" ]; then
+    kubectl config use-context "$KUBECONTEXT"
+fi
+
+if ! kubectl get namespace ${KUBE_NS} >/dev/null 2>/dev/null ; then
+    kubectl create namespace ${KUBE_NS}
+fi
+
+if [ -n "$INCLUDE_TEST_STAGE" ]; then
+    # The variable is not empty, so it has a value
+    echo "INCLUDE_TEST_STAGE is not empty. Its value is: $INCLUDE_TEST_STAGE"
+else
+    # The variable is empty, so assign a default value
+    INCLUDE_TEST_STAGE="lint,stress,fqltool"
+    echo "INCLUDE_TEST_STAGE is empty. Assigning default value: $INCLUDE_TEST_STAGE"
+fi
+
+
+TARGETS=".jenkins/job/DslJob.jenkins"
+REPO_BRANCH="infracloud/cassandra-5.0"
+REPO_URL="https://github.com/infracloudio/cassandra.git"
+
+
+sed -i -e "/targets:/s|:.*$|: \"$TARGETS\"|" \
+    -e "/repositoryBranch:/s|:.*$|: \"$REPO_BRANCH\"|" \
+    -e "/repositoryUrl:/s|:.*$|: \"$REPO_URL\"|" jenkins-deployment.yaml
+
+# Add Helm Jenkins Operator repository
+echo "Adding Helm repository for Jenkins Operator..."
+helm repo add --namespace ${KUBE_NS} jenkins https://raw.githubusercontent.com/jenkinsci/kubernetes-operator/master/chart
+
+# Install Jenkins Operator using Helm
+echo "Installing Jenkins Operator..."
+#helm upgrade --namespace ${KUBE_NS} --install jenkins-operator jenkins/jenkins-operator --set jenkins.enabled=false --set jenkins.backup.enabled=false --version 0.8.0-beta.2 
+
+echo "Jenkins Operator installed successfully!" # condition to check if above command was success
+
+# deploy jenkins Instance TODO jenkins file parameter
+kubectl apply --namespace ${KUBE_NS} -f ${CASSANDRA_DIR}/.build/jenkins-deployment.yaml
+
+TOKEN=$(kubectl  get secret jenkins-operator-credentials-example -o jsonpath="{.data.token}" | base64 --decode)
+
+# Trigger a new build and capture the response headers
+response_headers=$(curl -i -X POST http://localhost:8080/job/k8s-e2e/buildWithParameters -u jenkins-operator:$TOKEN --data-urlencode "TEST_STAGES_TO_RUN=$INCLUDE_TEST_STAGE" 2>&1)
```

Comment:
this won't work without the port-forwarding
```suggestion
nohup kubectl port-forward svc/jenkins-operator-http-example 8080:8080 &
response_headers=$(curl -i -X POST http://127.0.0.1:8080/job/k8s-e2e/buildWithParameters -u jenkins-operator:$TOKEN --data-urlencode "TEST_STAGES_TO_RUN=$INCLUDE_TEST_STAGE" 2>&1)
```

(also 127.0.0.1 is safer than localhost)

Reply:
updated^

Reply:
added and tested, the port-forwarding is working in background